python/sqlflow_submitter/xgboost/feature_column_test.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+import sqlflow_submitter.xgboost.feature_column as fc
+from sqlflow_submitter.xgboost.feature_column import hashing
+
+
+def get_hash(x, bucket_size):
+    if isinstance(x, np.ndarray):
+        ret = np.ndarray(x.shape, dtype='int64')
+        for i in range(x.size):
+            ret.put(i, hashing(x.take(i)) % bucket_size)
+    else:
+        ret = hashing(x) % bucket_size
+
+    return ret
+
+
+def build_one_hot_vector(idx, size):
+    ret = np.zeros([size], dtype='int64')
+    ret[idx] = 1
+    return ret
+
+
+class TestFeatureColumnBase(unittest.TestCase):
+    def check(self, column, column_names, inputs, expected_outputs):
+        if not isinstance(inputs, (list, tuple)):
+            inputs = (inputs, )
+
+        if not isinstance(expected_outputs, (list, tuple)):
+            expected_outputs = (expected_outputs, )
+
+        self.assertEqual(len(inputs), len(expected_outputs))
+
+        column = fc.ComposedColumnTransformer(column_names, column)
+        outputs = column((inputs, ))[0]
+
+        self.assertEqual(len(inputs), len(outputs))
+
+        if isinstance(outputs[0], np.ndarray):
+            if not np.array_equal(outputs[0], expected_outputs[0]):
```

Comment:
Can we move this print into self.assertTrue as the message of assertion failure?

Reply:
Sorry, this is the debug info when developing. I would remove it.